src/libraries/System.Text.Json/src/Polyfills/System.Text.Rune.netstandard20.cs

```diff
@@ -0,0 +1,547 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+
+// Contains a polyfill implementation of System.Text.Rune that works on netstandard2.0.
+// Implementation copied from:
+// https://github.com/dotnet/runtime/blob/177d6f1a0bfdc853ae9ffeef4be99ff984c4f5dd/src/libraries/System.Private.CoreLib/src/System/Text/Rune.cs
```

Comment:
Rather than duplicating this code, would it be feasible multi-target the original source file in corelib? It might be something to consider assuming we can keep the number of build conditionals tolerable.

Ideally we'd also want to run our existing test suite against the polyfill, but I can't think of a good strategy that doesn't involve IVT or creating an OOB package. cc @GrabYourPitchforks

Reply:
I updated the reference to link to the CoreLib version instead, but now we need to duplicate the `ThrowHelper` and resources. I think this is preferable to multi-targeting the CoreLib `ThrowHelper` and also better than duplicating `Rune`.